src/font.rs

```diff
@@ -353,6 +353,70 @@ impl Font {
             .guidelines
             .get_or_insert_with(Default::default)
     }
+
+    /// Returns a vector of immutable lib key string references
+    pub fn lib_keys(&self) -> Vec<&String> {
+        self.lib.keys().collect::<Vec<&String>>()
+    }
+
+    /// Returns an iterator over immutable lib key string references
+    pub fn iter_lib_keys(&self) -> impl Iterator<Item = &String> {
+        self.lib.keys()
+    }
+
+    /// Returns a vector of immutable group name string references
+    pub fn group_names(&self) -> Vec<&String> {
+        match &self.groups {
+            Some(g) => g.keys().collect::<Vec<&String>>(),
+            None => vec![],
+        }
+    }
+
+    /// Returns an iterator over immutable group name string references
+    pub fn iter_group_names(&self) -> impl Iterator<Item = &String> {
+        match &self.groups {
+            Some(g) => {
+                let mut groupvec = vec![];
+                for groupname in g.keys() {
+                    groupvec.push(groupname)
+                }
+                groupvec.into_iter()
+            }
+            None => vec![].into_iter(),
+        }
```

Comment:
It isn't very obvious, but there's a reasonable one-liner here
```suggestion
        self.groups.as_ref().into_iter().flat_map(|g| g.keys().map(String::as_ref))
```